File: comparison/results/haiku/ponytail/humaneval/HumanEval_1/go/solution.go
func separate_paren_groups(paren_string string) []string {
    var cleaned string
    for _, ch := range paren_string {
        if ch != ' ' {
            cleaned += string(ch)
        }
    }
    
    var result []string
    var current string
    var balance int
    
    for _, ch := range cleaned {
        current += string(ch)
        if ch == '(' {
            balance++
        } else if ch == ')' {
            balance--
        }
        
        if balance == 0 && current != "" {
            result = append(result, current)
            current = ""
        }
    }
    
    return result
}

File: comparison/results/haiku/ponytail/humaneval/HumanEval_1/java/solution.java
public static ArrayList<String> separateParenGroups(String paren_string) {
    ArrayList<String> result = new ArrayList<>();
    String cleaned = paren_string.replaceAll(" ", "");
    
    int balance = 0;
    StringBuilder current = new StringBuilder();
    
    for (char c : cleaned.toCharArray()) {
        current.append(c);
        if (c == '(') {
            balance++;
        } else if (c == ')') {
            balance--;
        }
        
        if (balance == 0 && current.length() > 0) {
            result.add(current.toString());
            current = new StringBuilder();
        }
    }
    
    return result;
}
